src/Symfony/Bundle/FrameworkBundle/composer.json

```diff
@@ -29,6 +29,7 @@
         "symfony/polyfill-mbstring": "~1.0",
         "symfony/filesystem": "^5.4|^6.0",
         "symfony/finder": "^5.4|^6.0",
+        "symfony/secret": "^6.1",
```

Comment:
IMHO this should be an optional dependency (The FrameworkBundle does not **require** secrets to work). 

But I'm not sure how we could provide a BC layer (migration path) without requiring it nor duplicating the code.

Reply:
can it be also enabled/disabled via a new framework config key?

Reply:
> can it be also enabled/disabled via a new framework config key?

That's already the case. 

The point of NOT making it a required dependency is to avoid downloading everything "just in case". Otherwise, `symfony/framework-bundle` will become the next `symfony/symfony`

Reply:
oki I got it thx for explanations :)

Reply:
Let's make it an optional dependency in 7.0 then? Otherwise we would need to duplicate the code and lose the advantage of the optional dependency in terms of download size.

Reply:
@derrabus agreed, but how triggering a deprecation if the user did not require the deps. Otherwise, bumping symfony to 7.0 will uninstall `symfony/secrets`

Reply:
Sounds good to me

Reply:
I think, we've had similar problems when we removed messenger transports into bridge packages. How did we solve this issue back then? 🤔

Reply:
We made messenger require the bridges (hard dep) until 6.0